clap_markdown/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+pub struct Markdown {
+    /// Name of the (sub)command and all its parents.
+    ///
+    /// The order of commands goes from topmost as the first element to the
+    /// command itself in the last position.
+    pub command: Vec<String>,
+    /// Generated markdown page.
+    pub content: String,
+}
+
+impl Markdown {
+    fn from_command(cmd: &mut Command, mut parents: Vec<String>) -> Self {
+        // Add (sub)command to command list.
+        parents.push(cmd.get_name().into());
+
+        let mut markdown = String::new();
+
+        // Add command description.
+        if let Some(description) =
+            cmd.get_long_about().or_else(|| cmd.get_about()).map(ToString::to_string)
+        {
+            let description = escape_markdown(&description);
+            let _ = writeln!(markdown, "{description}");
+        }
+
+        // Add usage example.
+        let usage = cmd.render_usage().to_string();
+        let _ = writeln!(markdown, "\n```sh\n{usage}\n```");
+
+        // Add positional arguments.
+        let mut positionals = cmd.get_positionals().flat_map(generate_argument).peekable();
+        if positionals.peek().is_some() {
+            let _ = writeln!(markdown, "\n### Arguments");
+        }
+        for positional in positionals {
+            let _ = writeln!(markdown, "\n{positional}");
+        }
+
+        // Add options.
+        let mut options = cmd
+            .get_arguments()
+            .filter(|arg| !arg.is_positional())
+            .flat_map(generate_argument)
+            .peekable();
+        if options.peek().is_some() {
+            let _ = writeln!(markdown, "\n### Options");
+        }
+        for option in options {
+            let _ = writeln!(markdown, "\n{option}");
+        }
+
+        // Add subcommands.
+        let mut subcommands = cmd
+            .get_subcommands()
+            .filter(|cmd| !cmd.is_hide_set() && cmd.get_name() != "help")
+            .peekable();
+        if subcommands.peek().is_some() {
+            let _ = writeln!(markdown, "\n### Commands\n");
+        }
+        for cmd in subcommands {
+            let name = cmd.get_name();
+            let human_path = format!("{} {name}", parents.join(" "));
+            let link_path = format!("{}_{name}", parents.join("_"));
+            let _ = writeln!(markdown, "* [{}](./{})", human_path, link_path);
+        }
+
+        Self { command: parents, content: markdown }
+    }
+}
+
+/// Convert argument to markdown.
+fn generate_argument(arg: &Arg) -> Option<String> {
+    // Don't show hidden arguments.
+    if arg.is_hide_set() {
+        return None;
+    }
+
+    let mut markdown = String::new();
+
+    // Add short option.
+    if let Some(short) = arg.get_short() {
+        let _ = write!(markdown, "-{short}");
+    }
+
+    // Add long option.
+    if let Some(long) = arg.get_long() {
+        if !markdown.is_empty() {
+            markdown += ", ";
+        }
+        let _ = write!(markdown, "--{long}");
```

Comment:
```suggestion
        let _ = write!(markdown, "`--{long}`");
```
To be more like the original (current) docs.